### ISSUE TITLE
[Bug]: Changing identifier in RDS Cluster and Cluster Instance (aws_rds_cluster & aws_rds_cluster_instance) forces recreate

### ISSUE THREAD

**Issue by DemonICOLI** [hashicorp/terraform-provider-aws#38644]
### Terraform Core Version

1.9

### AWS Provider Version

5.0

### Affected Resource(s)

* aws_rds_cluster
* aws_rds_cluster instance

### Expected Behavior

Neither of those resources should be force recreated as this change can be done no issues on AWS Console... this also seem to affect normal instances at some point, but fix was not propagated to cluster, [https://github.com/hashicorp/terraform-provider-aws/issues/507](url)

### Actual Behavior

Cluster and Cluster instances will have their identifiers updated without recreation

### Terraform Configuration Files

```

resource "aws_rds_cluster" "postgresql" {
  cluster_identifier      = "aurora-cluster-demo" <--- CHANGE THIS
  engine                  = "aurora-postgresql"
  availability_zones      = ["us-west-2a", "us-west-2b", "us-west-2c"]
  database_name           = "mydb"
  master_username         = "foo"
  master_password         = "must_be_eight_characters"
  backup_retention_period = 5
  preferred_backup_window = "07:00-09:00"
}
```

### Steps to Reproduce

Create an aurora cluster and cluster instance
try to rename them by use the identifier property

### Would you like to implement a fix?

None

**Comment by AnthonyGarriga** [comment]
I'm having the same issue. I simply want to change the identifier on the cluster and cluster instance and Terraform forces a recreate, which removes all data from the cluster. Is there any workaround?

**Comment by aviau** [comment]
Same issue here. It looks like workaround is to delete the resource from the state and re-import it...

**Comment by AnthonyGarriga** [comment]
> Same issue here. It looks like workaround is to delete the resource from the state and re-import it...

Yes, I ended up removing the resources from the state, renaming the RDS cluster and cluster instance in the AWS console, and then re-importing them to the state. It works for now but it's a workaround.

**Comment by justinretzolk** [comment]
Similar #34935
Related #38599

**Comment by MS99-9** [comment]
Hey @justinretzolk , If nobody is working on this I am happy to work on it 👍

**Comment by jerzygangi** [comment]
## Manual workaround for renaming a RDS cluster instance `identifier`
Until this bug is fixed, here's a manual workaround of how to rename a RDS cluster instance's `identifier`. Assuming you have a `aws_rds_cluster_instance` called `my_instance`, and you want to rename it's identifier to be `new-name`:

1. As a sanity check: run `terraform plan` and confirm no changes are pending application.
1. Within the RDS console, select the instance and click `Modify`.
1. Note the existing name of the `DB instance identifier`.
1. Rename the `DB instance identifier` to be the name you want (e.g., `new-name`.)
1. Save the change. (i.e., click `Continue` and then `Modify DB instance`).
1. Wait for the renaming to finish; when the instance's status is `Available`, it's done.
1. As a sanity check: run `terraform plan`, and confirm the RDS instance says it'll be created. (Don't apply, obviously.)
1. Remove the RDS instance from the Terraform state file:
    1. Optionally: create a backup of your state file.
    1. As a sanity check: run `terraform state rm -dry-run 'aws_rds_cluster_instance.my_instance'`, and confirm the instance would be deleted.
    1. Run `terraform state rm 'aws_rds_cluster_instance.my_instance'`.
1. Within your `resource "aws_rds_cluster_instance" "my_instance"` Terraform code block, set the `identifier` attribute to `new-name`.
1. Import the RDS instance back into your Terraform state: `terraform import aws_rds_cluster_instance.my_instance new-name`. (Assuming you've used `my_instance` and `new-name`, above.) The stdout should contain `Import successful!`.1.
1. As a sanity check: run `terraform plan` and confirm no changes are pending application.

Your cluster instance is now renamed.

(Confirmed on an Aurora Serverless v2 PostgreSQL cluster instance on February 1, 2025.)
